app/experimenter/experiments/tests/test_models.py

```diff
@@ -1155,21 +1155,56 @@ def test_rollouts_monitoring_dashboard_returns_correct_formatted_url(self):
             ),
         )
 
-    def test_rollouts_monitoring_dashboard_returns_url_when_rollout_is_complete(self):
+    @parameterized.expand(
+        [
+            (
+                False,
+                NimbusExperiment.Status.DRAFT,
+                NimbusExperimentFactory.Lifecycles.CREATED,
+            ),
+            (
+                True,
+                NimbusExperiment.Status.COMPLETE,
+                NimbusExperimentFactory.Lifecycles.ENDING_APPROVE_APPROVE,
+            ),
+            (
+                True,
+                NimbusExperiment.Status.LIVE,
+                NimbusExperimentFactory.Lifecycles.LIVE_ENROLLING,
+            ),
+            (
+                True,
+                NimbusExperiment.Status.LIVE,
+                NimbusExperimentFactory.Lifecycles.LAUNCH_APPROVE_APPROVE,
+            ),
+            (
+                False,
+                NimbusExperiment.Status.PREVIEW,
+                NimbusExperimentFactory.Lifecycles.PREVIEW,
+            ),
+        ]
+    )
+    def test_rollouts_monitoring_dashboard_returns_url(
+        self, valid_status, status, lifecycle
+    ):
         experiment = NimbusExperimentFactory.create_with_lifecycle(
-            NimbusExperimentFactory.Lifecycles.ENDING_APPROVE,
+            lifecycle=lifecycle,
             slug="rollout-1-slug",
             is_rollout=True,
-            status=NimbusExperiment.Status.COMPLETE,
-        )
-        expected_slug = "rollout_1_slug"
-        self.assertEqual(
-            experiment.rollout_monitoring_dashboard_url,
-            settings.ROLLOUT_MONITORING_URL.format(
-                slug=expected_slug,
-            ),
+            status=status,
         )
 
+        if valid_status:
+            expected_slug = "rollout_1_slug"
```

Comment:
The slug is hardcoded here twice, just move that up to the top of the test, or refer to `experiment.slug` here.

Reply:
It's a different formatted slug here. It's formatted from "-" to "_" (https://github.com/mozilla/experimenter/pull/7640#discussion_r958853787 hehe)

Reply:
I wanted to leave it outside of the `assert` so that it's clear that it's not just the normal slug as-is, but I'm open to just hardcoding it inside the `assert` since it's only used once. WDYT?

Reply:
Oh the conversion only goes one way, `-` to `_` so if you use the `_` version for both then they'll just match and nothing will get transformed, but then you're not really testing the conversion in the logic.  Yeah okay I see your point I forgot about the `-` `_` thing.  Then yeah this is fine.